templates/publish.yml

```diff
@@ -0,0 +1,118 @@
+# Copyright 2024 Defense Unicorns
+# SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+
+spec:
+  inputs:
+    runsOn:
+    flavor:
+    artifact_dir:
+      default: .ci_artifacts
+    reports-path:
+      default: ""
+    registry1Username:
+      default: ${IRON_BANK_ROBOT_USERNAME}
+    registry1Password:
+      default: ${IRON_BANK_ROBOT_PASSWORD}
+    chainguardIdentity:
+      default: ${CHAINGUARD_IDENTITY}
+    targetRepo:
+      default: ""
+    team:
+      default: uds
+---
+
+publish:
+  variables:
+    GIT_STRATEGY: clone
+    GIT_DEPTH: 0
+  id_tokens:
+    CHAINGUARD_TOKEN:
+      aud: https://console-api.enforce.dev
+  rules:
+    - if: $CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH
+    - if: $CI_MERGE_REQUEST_ID
+      variables:         # Override DRY_RUN defined
+        DRY_RUN: "true"  # at the job level.
+  script:
+    # Dry run on non-tag runs
+    - OPTIONS="--no-progress"
+    - if [[ "$DRY_RUN" == "true" ]]; then OPTIONS="$OPTIONS --dry-run"; fi
+
+    # Environment setup
+    - |
+      uds run actions:authenticate-registries --set REGISTRY1_USERNAME="$[[ inputs.registry1Username ]]" \
+                                              --set REGISTRY1_PASSWORD="$[[ inputs.registry1Password ]]" \
+                                              --set CHAINGUARD_IDENTITY="$[[ inputs.chainguardIdentity ]]" \
+                                              --set CHAINGUARD_TOKEN="$CHAINGUARD_TOKEN" \
+                                              --set GITLAB_REGISTRY_URL="$CI_REGISTRY" \
+                                              --set GITLAB_REGISTRY_USER="$CI_REGISTRY_USER" \
+                                              --set GITLAB_REGISTRY_TOKEN="$CI_REGISTRY_PASSWORD"
+
+    # Check if release is necessary for flavor and exit early if not
+    - |
+      if [[ "$DRY_RUN" == "true" ]]; then
+        echo "Dry run, skipping check if release is required: uds-releaser check ${FLAVOR}"
+      else
+        if ! uds-releaser check ${FLAVOR}; then echo "No release necessary"; exit 0; fi
+      fi
+
+    # Modify zarf package and uds bundle to have the correct version
+    - |
+      if [[ "$DRY_RUN" == "true" ]]; then
+        echo "Dry run, skipping release command: uds-releaser update-yaml ${FLAVOR}"
+      else
+        uds-releaser update-yaml ${FLAVOR}
+      fi
```

Comment:
if it is dry-running the publish wouldn't we still want to run these? They should run relatively quick (and it would be less logic to maintain)

Reply:
I think we can keep the `update-yaml` but the check flavor actually stops the processing if there is "No release necessary" so it actually keeps any of the other dry runs from happening. I'll test what happens if we keep `update-yaml` in there.

Reply:
Actually I can refactor the `check flavor` to still run but not exit if dry run is enabled.

Reply:
Refactored this so that the `uds-releaser check` still runs all the time but if dry run is on it doesn't `exit` so that other commands can actually still execute with the dry run flag.

Updated `uds-releaser update-yaml` to run all the time as-is because it is harmless.